### PR TITLE
Readiness indicator by file

### PR DESCRIPTION
**Pull request by AdamHawtin**
## Motivation and Context
Without some indicator of readiness, Kubernetes considers the app ready as soon as its container is running, even if it is not properly configured and crashes soon after. The main downside to this is that deployments from concourse may go green when the container starts, even if the app is misconfigured and crashes immediately afterwards. This PR adds a file indicator which is only created once the app has completed its setup and is entering its message polling.

## What has changed
- Add a readiness context manager which creates and removes an indicator file
- Use readiness context manager on the main message polling loop after successful start up
- Add new environment variable for readiness file path

## How to test?
- Run the app properly configured and it should start up and create the readiness file. Then when it is stopped the file should be removed.

- Run the app with broken config (e.g. an incorrect topic name), it should fail to start and never create the readiness file at any point.

## Links
https://trello.com/c/UDZtgxSZ/483-deploy-census-rm-pubsub-to-kubernetes-8

### CODE REVIEW

**Comment by AdamHawtin** [comment]
[deploy census-rm-pubsub to kubernetes (8)](https://trello.com/c/UDZtgxSZ/483-deploy-census-rm-pubsub-to-kubernetes-8)

**Comment by AdamHawtin** [comment]
[deploy census-rm-pubsub to kubernetes (8)](https://trello.com/c/UDZtgxSZ/483-deploy-census-rm-pubsub-to-kubernetes-8)

File: run.py [hunk 1]
```diff
@@ -21,9 +22,10 @@ def main():
     init_rabbitmq()  # test the connection to the rabbitmq cluster
 
     subscriber_future = setup_subscription()
-    while True:  # setup_subscription creates a background thread for processing messages
-        subscriber_future.result(timeout=None)  # block main thread while polling for messages indefinitely
-        time.sleep(30)
+    with Readiness(os.getenv('READINESS_FILE_PATH')):  # Indicate ready after successful setup
```
Comment (jcox-dev) [inline run.py]: Perhaps it's worth having a default here?

File: app/readiness.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        self.readiness_file = readiness_file
+        self._is_ready = False
+
+    def __enter__(self):
+        self.make_ready()
+
+    def __exit__(self, *args):
+        self.make_unready()
+
+    def make_ready(self):
```
Comment (jcox-dev) [inline app/readiness.py]: 👍 succinct

File: app/readiness.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import os
+
+
+class Readiness:
+    def __init__(self, readiness_file):
+        self.readiness_file = readiness_file
+        self._is_ready = False
+
+    def __enter__(self):
+        self.make_ready()
```
Comment (jcox-dev) [inline app/readiness.py]: Could have some simple (debug-level) logging here and `__exit__`

**Review by jcox-dev — COMMENTED** [review]
Just general comments but works as expected and is great that we're not introducing any further dependencies 👍

**Review by jcox-dev — APPROVED** [review]
👍

**Review by liamtoozer — APPROVED** [review]
Works for me locally 👍

**Pull request merged**